### PR TITLE
feat(components): [color-picker] add empty values attributes

### PR DESCRIPTION
**Pull request by Dsaquel**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

apply https://github.com/element-plus/element-plus/pull/20880#issuecomment-2939496871

### CODE REVIEW

File: packages/components/color-picker/__tests__/color-picker.test.tsx [hunk 1]
```diff
@@ -106,7 +106,7 @@ describe('Color-picker', () => {
     await wrapper.find('.el-color-picker__trigger').trigger('click')
     document.querySelector<HTMLElement>('.el-color-dropdown__btn')?.click()
     await nextTick()
-    expect(color.value).toEqual('') // should be empty #13239
+    expect(color.value).toBeNull()
```
Comment (btea) [inline packages/components/color-picker/__tests__/color-picker.test.tsx:110]: Perhaps we should add a test case that clears the value to an empty string.

**Review by btea — APPROVED** [review]

File: packages/components/color-picker/__tests__/color-picker.test.tsx (hunk 1 above)
Comment (Dsaquel) [reply packages/components/color-picker/__tests__/color-picker.test.tsx:110]: I agree, I'll add it later in the day.

File: packages/components/color-picker/src/color-picker.vue [hunk 2]
```diff
@@ -286,7 +288,7 @@ function handleConfirm() {
 }
 
 function confirmValue() {
-  const value = color.value
+  const value = isEmptyValue(color.value) ? valueOnClear.value : color.value
```
Comment (kooriookami) [inline packages/components/color-picker/src/color-picker.vue:292]:
```suggestion
  const value = color.value || valueOnClear.value
```
color.value can only be empty string when no color, so isEmptyValue couldn't work.

**Review by kooriookami — CHANGES_REQUESTED** [review]

File: packages/components/color-picker/src/color-picker.vue (hunk 2 above)
Comment (Dsaquel) [reply packages/components/color-picker/src/color-picker.vue:292]: You right, but when there is a color where we want to set as an empty values it will fallback to valueOnClear.

Comment (kooriookami) [reply packages/components/color-picker/src/color-picker.vue:292]: The empty value should be `valueOnClear`, that's seems no problem.

Comment (Dsaquel) [reply packages/components/color-picker/src/color-picker.vue:292]: If we want to includes color as empty values we should make the isEmptyValue(color.value) judgment, [demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtY29sb3ItcGlja2VyXG4gICAgdi1tb2RlbD1cImNvbG9yMVwiXG4gICAgOmVtcHR5LXZhbHVlcz1cImVtcHR5VmFsdWVzXCJcbiAgICA6dmFsdWUtb24tY2xlYXI9XCJudWxsXCJcbiAgLz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYsIHdhdGNoRWZmZWN0IH0gZnJvbSAndnVlJ1xuXG5jb25zdCBjb2xvcjEgPSByZWYoJyNmZmYnKVxuY29uc3QgZW1wdHlWYWx1ZXMgPSBbJyMwMDAwMDAnLCBudWxsLCB1bmRlZmluZWRdXG53YXRjaEVmZmVjdCgoKSA9PiBjb25zb2xlLmxvZygnY29sb3IxJywgY29sb3IxLnZhbHVlKSlcbi8vMTogc2V0IGluIHRoZSBpbnB1dCAjMDAwXG4vLzI6IGNvbmZpcm1cbi8vMzogY29sb3IxIGlzIG5vdyBudWxsXG48L3NjcmlwdD5cblxuPHN0eWxlPlxuLmRlbW8tY29sb3ItYmxvY2sge1xuICBkaXNwbGF5OiBmbGV4O1xuICBhbGlnbi1pdGVtczogY2VudGVyO1xuICBtYXJnaW4tYm90dG9tOiAxNnB4O1xufVxuLmRlbW8tY29sb3ItYmxvY2sgLmRlbW9uc3RyYXRpb24ge1xuICBtYXJnaW4tcmlnaHQ6IDE2cHg7XG59XG48L3N0eWxlPlxuIiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIGNvbnN0IHN0eWxlcyA9IFsnaHR0cHM6Ly9wcmV2aWV3LTIwOTYyLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvZGlzdC9pbmRleC5jc3MnLCAnaHR0cHM6Ly9wcmV2aWV3LTIwOTYyLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvdGhlbWUtY2hhbGsvZGFyay9jc3MtdmFycy5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbFNldHRsZWQoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTIwOTYyLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvZGlzdC9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTIwOTYyLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvZGlzdC9pbmRleC5jc3MifX0=).

Might be overkill/not really convenient tho. wdyt ?

Comment (kooriookami) [reply packages/components/color-picker/src/color-picker.vue:292]: Oh, no. `emptyValues` should be a falsy value. `valueOnClear` only accept falsy value. @btea

Comment (btea) [reply packages/components/color-picker/src/color-picker.vue:292]: According to your discussion above, maybe we should not move the logic related to empty value here, and it is better to keep the status quo.

**Review by kooriookami — APPROVED** [review]

**Review by warmthsea — APPROVED** [review]
LGTM
